lldb/include/lldb/Core/Telemetry.h

```diff
@@ -66,6 +79,42 @@ struct LLDBBaseTelemetryInfo : public llvm::telemetry::TelemetryInfo {
   void serialize(llvm::telemetry::Serializer &serializer) const override;
 };
 
+struct CommandInfo : public LLDBBaseTelemetryInfo {
+
+  // If the command is/can be associated with a target entry this field contains
+  // that target's UUID. <EMPTY> otherwise.
+  std::string target_uuid;
+  // A unique ID for a command so the manager can match the start entry with
+  // its end entry. These values only need to be unique within the same session.
+  // Necessary because we'd send off an entry right before a command's execution
+  // and another right after. This is to avoid losing telemetry if the command
+  // does not execute successfully.
+  int command_id;
+  // Eg., "breakpoint set"
+  std::string command_name;
+  // !!NOTE!! These two fields are not collected by default due to PII risks.
+  // Vendor may allow them by setting the
+  // LLDBConfig::m_detailed_command_telemetry.
+  std::string original_command;
+  std::string args;
```

Comment:
Oh I see, the `m_detailed_command_telemetry` boolean controls a subset of the command, which means my earlier suggestion doesn't work. I can't think of a way to make this more "generic" without breaking up the event, which you probably don't want to do?

Reply:
Right - that was the knob for controlling how much data is collected upstream.

(The previous design was to have those `DispatchFoo()` helper method which could be overriden by vendors to collect more or less data, as needed. There's pros and cons to each ...)